src/instructlab/data/generate.py

```diff
@@ -210,6 +221,13 @@ def generate(
         except Exception as exc:
             click.secho(f"Failed to start server: {exc}", fg="red")
             raise click.exceptions.Exit(1)
+
+        # disable batching when running with the local llama.cpp server
+        if isinstance(backend_instance, llama_cpp_server):
+            batch_size = 0
+            logger.warning(
+                "Disabling SDG batching - unsupported with llama.cpp serving"
```

Comment:
Should we only log this message when the user requested a particular batch-size?

Reply:
Yep, good point